src/ctrl/PopupMenuParamWidget.h

```diff
@@ -135,11 +164,21 @@ class PopupMenuItem : public ::rack::ui::MenuItem {
         parent->text = this->text;
         ::rack::event::Change ce;
         if (parent->getParamQuantity()) {
-            float value = index;
+            float newValue = index;
+            const float oldValue =  parent->getParamQuantity()->getValue();
             if (parent->optionalIndexToValueFunction) {
-                value = parent->optionalIndexToValueFunction(index);
+                newValue = parent->optionalIndexToValueFunction(index);
             }
-            parent->getParamQuantity()->setValue(value);
+            parent->getParamQuantity()->setValue(newValue);
```

Comment:
widgets need to make the undo events, which I didn't realize until someone pointed out that this thing never does undo correctly.